src/renderer/components/layout/main-layout.tsx

```diff
@@ -18,43 +19,87 @@ interface Props {
 
 @observer
 export class MainLayout extends React.Component<Props> {
-  public storage = createStorage("main_layout", { pinnedSidebar: true });
+  public storage = createStorage("main_layout", {
+    pinnedSidebar: true,
+    sidebarWidth: 200,
+  });
 
   @observable isPinned = this.storage.get().pinnedSidebar;
   @observable isAccessible = true;
+  @observable sidebarWidth = this.storage.get().sidebarWidth
 
   @disposeOnUnmount syncPinnedStateWithStorage = reaction(
     () => this.isPinned,
     (isPinned) => this.storage.merge({ pinnedSidebar: isPinned })
   );
 
+  @disposeOnUnmount syncWidthStateWithStorage = reaction(
+    () => this.sidebarWidth,
+    (sidebarWidth) => this.storage.merge({ sidebarWidth })
+  );
+
   toggleSidebar = () => {
     this.isPinned = !this.isPinned;
     this.isAccessible = false;
     setTimeout(() => (this.isAccessible = true), 250);
   };
 
+  getSidebarSize = () => {
+    return {
+      "--sidebar-width": `${this.sidebarWidth}px`,
+    }
+  }
+
+  @autobind()
+  adjustWidth(newWidth: number): void {
+    this.sidebarWidth = newWidth
+  }
+
+  sidebarMinWidth(): number {
+    const headerPadding = 4 * 2 // 4px * (1 for right + 1 for left)
+    const appIcon = 28
+    const appName = 40
+    const pinningIcon = 22 + 10 // 10 for hover circle
+    const goodMeasure = 10
+    return headerPadding + appIcon + appName + pinningIcon + goodMeasure
+  }
```

Comment:
If we're using hard-coded numbers here so why not use just one? It's better so set min/max values inside css file
```
min-width: 170px;
max-width: 400px;
```
But keep in mind this would need fixing `--sidebar-width` value to keep it within min/max range. It can be done by checking sidebar's `getBoundingClientRect()` right after `onDragEnd` to adjust `--sidebar-width` with real value.

Reply:
This is exactly how the max height is calculated for the dock. I liked that style (so the number has some meaning as to where is came from)

Reply:
As to using css for min/max width. I would say it is better to have all the information in one place and since it is JS controlling the width that information should be also in the JS files.

Reply:
When we'll change sidebar layout like adding/removing elements on it, it's easy to forget to change this lines in `main-layout`. They can easily be outdated and separated from reality. And I think one single number already has a meaning - we understand that it comes from same calculations and offers readable size for most elements.

Reply:
okay, sure